### DISCUSSION TITLE
[firebase_firestore] useEmulator request

### DISCUSSION THREAD

**Discussion by radvansky-tomas** [firebase/flutterfire#6057] [Feature request] [answered]
I have issue using firebase firestore with hot reload and thats because we have to use .settings and not useEmulator as with other packages such as auth / functions.

Could we add / change this behaviour so it follows official SDKs:

https://firebase.google.com/docs/emulator-suite/connect_firestore#android_ios_and_web_sdks

```
// Firebase previously initialized using firebase.initializeApp().
var db = firebase.firestore();
if (location.hostname === "localhost") {
  db.useEmulator("localhost", 8080);
}
```

**Comment by Salakar** [comment]
cc @rrousselGit @russellwheatley - this would be a breaking change (though we can keep Settings with a deprecation decorator) but also is something I think we should do to match the other FlutterFire APIs from Auth & Functions (useEmulator) - changing it would also match the latest Firebase Web/Android/iOS naming too.

Here's the docs on how it does it currently; https://firebase.flutter.dev/docs/firestore/usage#emulator-usage vs Auth; https://firebase.flutter.dev/docs/auth/usage#emulator-usage vs Functions; https://firebase.flutter.dev/docs/functions/usage#emulator-usage

We should also probably deprecate the Functions one (`useFunctionsEmulator`) as that should also be changed to `useEmulator`. The APIs across all three are different (different name or args and also named vs positional etc) - I think we should standardise the names and the arguments.

**Comment by Salakar** [comment, marked as answer]
Tracking on https://github.com/FirebaseExtended/flutterfire/issues/5224
